gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/data/repository/AvatarRepository.kt

```diff
@@ -4,30 +4,42 @@ import android.net.Uri
 import androidx.core.net.toFile
 import com.gravatar.quickeditor.data.models.QuickEditorError
 import com.gravatar.quickeditor.data.storage.TokenStorage
+import com.gravatar.quickeditor.ui.avatarpicker.copy
 import com.gravatar.restapi.models.Avatar
 import com.gravatar.services.AvatarService
+import com.gravatar.services.ErrorType
 import com.gravatar.services.GravatarResult
 import com.gravatar.types.Email
 import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
 import kotlinx.coroutines.withContext
 
 internal class AvatarRepository(
     private val avatarService: AvatarService,
     private val tokenStorage: TokenStorage,
     private val dispatcher: CoroutineDispatcher,
 ) {
+    fun getAvatarsFlow(email: Email) = avatarsFlow(email).asSharedFlow()
+
+    private val avatarsFlows = mutableMapOf<String, MutableSharedFlow<EmailAvatars>>()
```

Comment:
The `EmailAvatars` object was created based on how the initial API was structured but also to serve the presentation layer. I don't think this is how we should store it. The `Avatar` object already has the `selected` field so it should be enough to store a plain list of Avatars and mapped to the `EmailAvatars` in the ViewModel.